golos.publication/golos.publication.cpp

```diff
@@ -210,21 +224,13 @@ void publication::create_message(structures::mssgid message_id,
         item.level = level;
     });
 
-    structures::accandvalue parent {parent_id.author, parent_pk};
-    uint64_t seconds_diff = 0;
-    while (parent.account) {
-        tables::message_table parent_message_table(_self, parent.account.value);
-        auto parent_itr = parent_message_table.find(parent.value);
-        eosio_assert(parent_itr != parent_message_table.end(), "parent message not found");
-        eosio_assert(cur_time >= parent_itr->date, "publication::create_message: cur_time < parent_itr->date");
-        seconds_diff = cur_time - parent_itr->date;
-        parent.account = parent_itr->parentacc;
-        parent.value = parent_itr->parent_id;
-    }
-    seconds_diff /= eosio::seconds(1).count();
-    uint64_t delay_sec = cashout_window_param.window > seconds_diff ? cashout_window_param.window - seconds_diff : 0;
-    if (delay_sec)
-        close_message_timer(message_id, message_pk, delay_sec);
+    structures::archive_info_v1 info{message_id,level};
+    uint64_t record_id = eosio::save_record(structures::archive_record{info});
+
+    structures::create_event evt{record_id};
+    eosio::event(_self, "postcreate"_n, evt).send();
```

Comment:
there is `send_xxx_event` section exists near the end of this file, propose move this code there to have similar code in one place